### PR TITLE
Simplify resource name handling

### PR DESCRIPTION
**Pull request by dbatyai**

### CODE REVIEW

**Comment by dbatyai** [comment]
Benchmark | Stack (bytes) | Perf (sec) |
----: | ----: | ----: | 
3d-cube.js | 16768 -> 16192 : **+3.435%** | 1.146 -> 1.125 : **+1.791%** | 
3d-morph.js | 1592 -> 1584 : +0.503% | 1.179 -> 1.154 : **+2.064%** | 
3d-raytrace.js | 2908 -> 2852 : **+1.926%** | 1.371 -> 1.351 : **+1.476%** | 
access-binary-trees.js | 3824 -> 3808 : +0.418% | 0.817 -> 0.780 : **+4.455%** | 
access-fannkuch.js | 1664 -> 1656 : +0.481% | 2.623 -> 2.595 : **+1.071%** | 
access-nbody.js | 1768 -> 1760 : +0.452% | 1.595 -> 1.589 : +0.404% | 
bitops-3bit-bits-in-byte.js | 1616 -> 1608 : +0.495% | 0.749 -> 0.723 : **+3.382%** | 
bitops-bits-in-byte.js | 1616 -> 1608 : +0.495% | 0.996 -> 0.977 : **+1.865%** | 
bitops-bitwise-and.js | 1440 -> 1448 : -0.556% | 1.227 -> 1.224 : +0.246% | 
bitops-nsieve-bits.js | 1616 -> 1608 : +0.495% | 1.530 -> 1.523 : +0.470% | 
controlflow-recursive.js | 73816 -> 73808 : +0.011% | 0.553 -> 0.517 : **+6.495%** | 
crypto-aes.js | 2456 -> 2412 : **+1.792%** | 0.917 -> 0.901 : **+1.749%** | 
crypto-md5.js | 1952 -> 1928 : **+1.230%** | 0.782 -> 0.763 : **+2.515%** | 
crypto-sha1.js | 1720 -> 1712 : +0.465% | 0.801 -> 0.779 : **+2.775%** | 
crypto.js | 3624 -> 3568 : **+1.545%** | 8.938 -> 8.825 : **+1.266%** | 
date-format-tofte.js | 2136 -> 2112 : **+1.124%** | 1.019 -> 0.997 : **+2.172%** | 
date-format-xparb.js | 2752 -> 2736 : +0.581% | 0.838 -> 0.829 : **+1.069%** | 
deltablue.js | 4196 -> 4132 : **+1.525%** | 5.867 -> 5.697 : **+2.894%** | 
math-cordic.js | 1616 -> 1608 : +0.495% | 1.733 -> 1.702 : **+1.765%** | 
math-partial-sums.js | 1584 -> 1576 : +0.505% | 1.056 -> 1.040 : **+1.563%** | 
math-spectral-norm.js | 1616 -> 1608 : +0.495% | 0.765 -> 0.750 : **+1.990%** | 
raytrace.js | 5672 -> 5400 : **+4.795%** | 2.912 -> 2.983 : -2.446% | 
richards.js | 2540 -> 2452 : **+3.465%** | 0.324 -> 0.308 : **+5.117%** | 
string-base64.js | 1648 -> 1640 : +0.485% | 1.437 -> 1.410 : **+1.837%** | 
string-fasta.js | 1600 -> 1592 : +0.500% | 1.851 -> 1.820 : **+1.685%** | 
Geometric mean: | +1.094% | +2.001% | 

Binary (bytes) | master(1b2e0aec8c) | patch(0a9e99f1ce) | Diff |
----: | ----: | ----: | ----: |
.rodata | 17143 | 17135 | -8 bytes |
.text | 190512 | 190320 | -192 bytes |

File: jerry-core/api/jerry.c [hunk 1]
```diff
@@ -3546,16 +3530,13 @@ jerry_get_backtrace (uint32_t max_depth) /**< depth limit of the backtrace */
 jerry_value_t
 jerry_get_resource_name (const jerry_value_t value) /**< jerry api value */
 {
-#if ENABLED (JERRY_RESOURCE_NAME)
```
Comment (zherczeg) [inline jerry-core/api/jerry.c:3549]: Are these macros removed? I see them in other files.

File: jerry-core/ecma/operations/ecma-function-object.c [hunk 2]
```diff
@@ -367,87 +352,77 @@ ecma_op_create_dynamic_function (const ecma_value_t *arguments_list_p, /**< argu
...
+    return ECMA_VALUE_ERROR;
+  }
 
 #if ENABLED (JERRY_ESNEXT)
-    ecma_value_t *func_name_p;
-    func_name_p = ecma_compiled_code_resolve_function_name ((const ecma_compiled_code_t *) bytecode_data_p);
-    *func_name_p = ecma_make_magic_string_value (LIT_MAGIC_STRING_ANONYMOUS);
+  ecma_value_t *func_name_p;
+  func_name_p = ecma_compiled_code_resolve_function_name ((const ecma_compiled_code_t *) bytecode_p);
+  *func_name_p = ecma_make_magic_string_value (LIT_MAGIC_STRING_ANONYMOUS);
```
Comment (zherczeg) [inline jerry-core/ecma/operations/ecma-function-object.c:376]: What is the difference between LIT_MAGIC_STRING_RESOURCE_ANON and LIT_MAGIC_STRING_ANONYMOUS? Why the former uses abbreviation?

File: jerry-core/parser/js/js-parser.c [hunk 3]
```diff
@@ -62,6 +62,12 @@ parser_compute_indicies (parser_context_t *context_p, /**< context */
   uint16_t ident_count = 0;
   uint16_t const_literal_count = 0;
 
+#if ENABLED (JERRY_RESOURCE_NAME)
+  /* Resource name will be stored as the last const literal. */
+  const_literal_count++;
+  context_p->literal_count++;
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser.c:68]: There is a limit for literals (32K or something), perhaps that check should be added here.

File: jerry-core/parser/js/js-parser.c [hunk 4]
```diff
@@ -2898,14 +2904,14 @@ parser_parse_script (const uint8_t *arg_list_p, /**< function argument list */
       /* It is unlikely that memory can be allocated in an out-of-memory
        * situation. However, a simple value can still be thrown. */
       jcontext_raise_exception (ECMA_VALUE_NULL);
-      return ECMA_VALUE_ERROR;
+      return NULL;
     }
 
     if (parser_error.error == PARSER_ERR_INVALID_REGEXP)
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser.c:2910]: This `if` and the `if` above sems to have the same body. Can they be merged?

File: jerry-core/api/jerry.c (hunk 1 above)
Comment (dbatyai) [reply jerry-core/api/jerry.c:3549]: I only removed them from here. Although I guess they should be kept to reduce code size.

File: jerry-core/ecma/operations/ecma-function-object.c (hunk 2 above)
Comment (dbatyai) [reply jerry-core/ecma/operations/ecma-function-object.c:376]: LIT_MAGIC_STRING_RESOURCE_ANON is '<anonymous>', and is used in resource names, while LIT_MAGIC_STRING_ANONYMOUS is 'anonymous' and is used in function names.

Comment (zherczeg) [reply jerry-core/ecma/operations/ecma-function-object.c:376]: So it is the same as empty? Perhaps we could use a macro alias then?

Comment (dbatyai) [reply jerry-core/ecma/operations/ecma-function-object.c:376]: Sorry, it's not empty, its `'<anonymous>'` (github thought it was markdown syntax, and removed it).

File: jerry-core/parser/js/js-parser.c (hunk 4 above)
Comment (dbatyai) [reply jerry-core/parser/js/js-parser.c:2910]: They seem distinct to me. In the first case were raising a new exception, while in the latter the context already contains one. I guess the return statement could be factored out though.

Comment (zherczeg) [reply jerry-core/parser/js/js-parser.c]: It was my mistake. Leave them as they are now.

**Pull request merged**